### ISSUE TITLE
Bad line wrapping in code blocks with PDF output

### ISSUE THREAD

**Issue by hlandau** [DaveJarvis/KeenWrite#163]
**Description**
When using the PDF output function, long lines in code blocks do not wrap correctly.

**Replicate**
See attached Markdown file:
- [untitled.md](https://github.com/DaveJarvis/KeenWrite/files/10610809/untitled.md)

Any code block with a long sequence of characters like:

```c
  // XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
```

will suffice to reproduce the issue.

The Solare theme was used.

**Expected**
Lines should be broken to keep them inside the body area of the page.

**Actual**
Characters run outside of the main area of the page in which body text should be constrained, and even off the paper.

- [untitled.pdf](https://github.com/DaveJarvis/KeenWrite/files/10610812/untitled.pdf)

**Screenshots**
Add screenshots to show the problem, if applicable.
![Screenshot_2023-02-05_11-24-51](https://user-images.githubusercontent.com/2270010/216815976-e62c6a1f-3690-4dc6-8eaf-fee9d7d34d81.png)

**Environment**
 - Operating System: Windows
 - Application: 3.2.0, freshly installed

**Comment by DaveJarvis** [comment]
> long lines in code blocks

Long lines signal potential problems with the code. Empirical studies have shown that line length is the most significant source code characteristic affecting readability across almost all programming languages. Whitespace and natural language identifier content are important influences on the readability of Java and Python, in particular.

Researchers found that **longer lines** make finding the beginning of the next line more difficult and **shorter lines** require tracking back to the next line more often. Both cases (too narrow or too wide) cause fatigue, implying that a happy middle must be found.

Further, there are practical reasons to impose shorter lines on source code.

* **Complexity.** Line length is primarily a function of nesting depth, indentation, and identifier names. Restraining line lengths inherently limits the other factors, which curtails complexity.
* **Reviews.** Diffs often use two side-by-side panels for code comparison. Code 120 columns wide, requires 240 columns, leading to horizontal scrolling depending on screen resolution, font size, or window size. Three-way merges would need 360 columns. The introduction of horizontal scrolling increases the amount of time needed to ensure the code is correct.
* **Media.** Code having shorter widths can be projected, printed, emailed, inserted into web pages, sent as text messages, **_inserted into PDF files_**, and relayed in other media without arbitrarily breaking lines. Long lines can affect screen sharing because people watching may be using different screen resolutions.
* **Scrolling.** Horizontal scrolling is tedious when reading text, which adds delays, interrupts pacing, and thus can negatively impact source code comprehension.
* **Visual acuity.** Some developers require larger font sizes due to visual impairment, which decreases the amount of horizontal real estate available for reading or reviewing code.
* **Historical.** Consistent column width restrictions for source code developed by different teams, languages, and projects over many years can also help reduce horizontal scrolling, which is a convenience that may help with the pace of reading code, and thereby affecting code comprehension.

Although few studies have focused on line length with respect to code comprehension, numerous studies and typographers discuss line length in printed text. Like printed text, line length on digital displays affects reading speed and comprehension. Research suggests using **shorter lines for accuracy**.

In _Program indentation and comprehensibility_ (1983), the authors found that indentation aids program comprehension and suggest that the optimal level of indentation is 2 to 4 spaces. My personal preference is 2 spaces of indentation, which leads to shorter lines.

In 2008, Buse and Weimer confirmed the conventional wisdom that programmers should use short lines, even if it means breaking up a statement across multiple lines. The following table lists various line length findings and recommendations for print and digital contexts:

| Length | Who | Source | Year |
|---|---|---|---|
| 45 - 75 | R. Bringhurst | [Elements of Typographic Style](https://readings.design/PDF/the_elements_of_typographic_style.pdf) | 1992 |
| 52 | K. Rayner | The Psychology of Reading | 1989 |
| 55 | M. Dyson | [The Influence of Reading Speed and Line Length on the Effectiveness of Reading From Screen](https://doi.org/10.1006/ijhc.2001.0458) | 2001 |
| 70 | H. Spencer | [The Visible Word](https://www.worldcat.org/title/visible-word/oclc/16472) | 1968 |
| 70 | A. Nanavati | Optimal Line Length in Reading --- A Literature Review | 2005 |
| 75 | R. Duchnicky | [Readability of Text Scrolled on Visual Display Terminals as a Function of Window Size](https://doi.org/10.1177%2F001872088302500605) | 1983 |
| 79 | R. Stallman | [GNU Formatting Style Guide](https://www.gnu.org/prep/standards/html_node/Formatting.html) | 2021 |
| 80 | M. Cooper | [Guide to Visual Presentation](https://www.w3.org/TR/UNDERSTANDING-WCAG20/visual-audio-contrast-visual-presentation.html) | 2016 |
| 95 | D. Shaikh | [Effects of Line Length on Reading Performance of Online News Articles](https://doi.org/10.1177%2F154193120504900514) | 2005 |
| 100 | M. Dyson | [Effects of Line Length and Method of Movement on Patterns of Reading from Screen](https://www.proquest.com/openview/49f4fd2296eb9ecef039e3cd3921fe0b/1?pq-origsite=gscholar&cbl=1821103) | 1998 |

There's a lot more I could write on this subject.

In summary, keep the lines to 80 columns.
